src/fastoad/cmd/doe.py

```diff
@@ -0,0 +1,226 @@
+"""Tools for generating and postprocess Design of Experiments (DOEs) in FAST-OAD"""
+#  This file is part of FAST-OAD : A framework for rapid Overall Aircraft Design
+#  Copyright (C) 2025 ONERA & ISAE-SUPAERO
+#  FAST is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from dataclasses import dataclass, field
+from os import PathLike
+from typing import ClassVar, Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import smt.sampling_methods as samp
+from smt.applications.mfk import NestedLHS
+
+import fastoad.api as oad
+from fastoad._utils.files import as_path
+from fastoad.openmdao.variables import VariableList
+
+
+@dataclass
+class DOEVariable:
+    """
+    Represents a Design of Experiments (DOE) variable defined by its unique `id_variable`.
+
+    :param name: The name of the FAST-OAD DOE variable.
+    :param bound_lower: The lower bound of the variable. Defaults to None.
+    :param bound_upper: The upper bound of the variable. Defaults to None.
+    :param reference_value: A reference value used to adjust bounds as percentages. If given, `bound_lower` and
+                            `bound_upper` are considered percentages. Defaults to None.
+    :param bind_variable_to: Another DOEVariable instance to bind this variable to. When bound, this variable
+                             inherits the bounds of the bound variable, and no new ID is assigned.
+    :param name_pseudo: An optional alias for the variable. If not provided, it defaults to the value of `name`.
+    """
+
+    name: str
+    bound_lower: Optional[float] = None
+    bound_upper: Optional[float] = None
+    reference_value: Optional[float] = None
+    bind_variable_to: Optional["DOEVariable"] = None
+    name_pseudo: Optional[str] = None
+
+    # Class-level counters
+    _instance_counter: ClassVar[int] = 0
+    _next_instance: ClassVar[int] = -1  # The ID of the first var is 0
+
+    def __post_init__(self):
+        type(self)._instance_counter += 1
+        type(self)._next_instance += 1
+        if not self.name_pseudo:  # if no pseudo is given, pseudo=name
+            self.name_pseudo = self.name
+        if self.bind_variable_to:
+            self.id_variable = (
```

Comment:
For readability, please declare all instance attributes in the data class header. Non-init attributes can be declared using [field](https://docs.python.org/3/library/dataclasses.html#dataclasses.field).